util/dvsim/SimCfg.py

```diff
@@ -640,10 +600,32 @@ def _gen_results(self, run_results):
             self.results_summary["Name"] = self._get_results_page_link(
                 self.results_summary["Name"])
 
-        # Append failures for triage
-        if results.fail_msgs:
-            fail_msgs = "\n## List of Failures\n" + ''.join(results.fail_msgs)
-            results_str += fail_msgs
+        # Append bucketized failures for triage, sorted by descending number
+        # of failures.
+        if results.buckets:
+            self.errors_seen = True
+            by_tests = sorted(results.buckets.items(),
+                              key=lambda i: len(i[1]),
+                              reverse=True)
+            fail_msgs = ["\n## Failure Buckets", ""]
+            for bucket, tests in by_tests:
+                fail_msgs.append(f"*   ```{bucket}```:")
+                # The tests could be sorted by ascending wall clock time.
+                for count, (test, line) in enumerate(tests):
+                    if count == _MAX_TESTS_PER_BUCKET:
+                        fail_msgs.append(
+                            f"    *   ... {len(tests) - count} more tests.")
+                        break
+                    fail_msgs.append(
+                        f"    *   {test.qual_name}<br>")
+                    if line:
+                        fail_msgs.append(
+                            f"        line {line}, {test.get_log_path()}")
+                    else:
+                        fail_msgs.append(
+                            f"        {test.get_log_path()}")
```

Comment:
nit: perhaps this is a bit neater
Plus also, take care of the case where line == 0 which is also falsy. 
```python
                    log_line = f"    {test.get_log_path()}"
                    if line is not None:
                        log_line += f", line {line}"
                    fail_msgs.append(log_line)
```

Reply:
The first line should be line 1. I fixed this in Launcher._check_status(), where we were using the enumerate count, which starts at 0. Also, putting the line number after a possibly long log path is not as friendly as putting it first in the line.

Reply:
Hmm, ok. Perhaps capitalize `Line {line}` on 622 if it is printed first before the log path?

Reply:
Done.